--- origsrc/mac-3.99-u4-b5/configure.in	2006-06-01 07:08:18.000000000 -0500
+++ src/mac-3.99-u4-b5/configure.in	2006-08-02 17:49:24.394625000 -0500
@@ -70,7 +70,7 @@
 
 AC_MSG_CHECKING([for some Win32 platform])
 case "$host" in
-  *-*-mingw*|*-*-cygwin*)
+  *-*-mingw*)
     os_win=yes
     ;;
   *)
--- origsrc/mac-3.99-u4-b5/src/MACLib/APETag.cpp	2006-06-01 04:00:57.000000000 -0500
+++ src/mac-3.99-u4-b5/src/MACLib/APETag.cpp	2006-08-06 20:05:25.196000000 -0500
@@ -4,6 +4,37 @@
 #include "CharacterHelper.h"
 #include "IO.h"
 #include IO_HEADER_FILE
+#include <ctype.h>
+
+#ifdef __CYGWIN__
+/*
+ * Cygwin/GCC doesn't support wcstol/wcstod directly 
+ * patch borrowed from SwiProlog:
+ * http://gollem.swi.psy.uva.nl/twiki/pl/pub/Porting/SwiPrologOnCygwinX11/pl-5.6.4_on_cygwin-1.5.19_with_x11.patch
+ */
+#define wcstol my_wcstol
+_BEGIN_STD_C
+long my_wcstol (wchar_t* wsNumber, wchar_t** pwsAfterNumber, int nBase)
+{
+	long result;
+	char mbsNumber[132];
+	char *mbsAfter;
+	wchar_t *wsFirst;
+	char mbC[8];
+
+	wsFirst = wsNumber;
+	do { wctomb(mbC, *wsFirst); } while ( isspace(mbC[0]) && ++wsFirst );
+
+	wcstombs(mbsNumber, wsFirst, 130);
+
+	result = strtol(mbsNumber, &mbsAfter, nBase);
+
+	*pwsAfterNumber = (wchar_t *)(wsFirst + (mbsAfter - mbsNumber));
+
+	return result;
+}
+_END_STD_C
+#endif
 
 /*****************************************************************************************
 CAPETagField
--- origsrc/mac-3.99-u4-b5/src/MACLib/Assembly/Makefile.am	2006-06-01 04:00:56.000000000 -0500
+++ src/mac-3.99-u4-b5/src/MACLib/Assembly/Makefile.am	2006-08-06 18:49:33.802250000 -0500
@@ -1,5 +1,7 @@
 AUTOMAKE_OPTIONS = gnu
 
+INCLUDES = -I$(top_srcdir)/src/Shared
+
 noinst_LTLIBRARIES = libmacasm.la
 
 COMMON_SOURCES = \
--- origsrc/mac-3.99-u4-b5/src/MACLib/MD5.h	2006-06-01 04:00:57.000000000 -0500
+++ src/mac-3.99-u4-b5/src/MACLib/MD5.h	2006-08-06 19:00:40.850875000 -0500
@@ -19,7 +19,7 @@
 #ifndef MD5SUM_MD5_H
 #define MD5SUM_MD5_H
 
-typedef unsigned int uint32_t;
+typedef long unsigned int uint32_t;
 typedef unsigned char uint8_t;
 
 /*
--- origsrc/mac-3.99-u4-b5/src/MACLib/Makefile.am	2006-06-01 04:00:58.000000000 -0500
+++ src/mac-3.99-u4-b5/src/MACLib/Makefile.am	2006-08-02 17:47:16.175875000 -0500
@@ -12,6 +12,7 @@
 
 libmac_la_LIBADD = $(shared_ldadd) $(asm_ldadd)
 libmac_la_LDFLAGS = \
+	-no-undefined \
 	-export-dynamic \
 	-version-info 2:0:0
 
--- origsrc/mac-3.99-u4-b5/src/Shared/StdLibFileIO.cpp	2006-06-01 04:00:56.000000000 -0500
+++ src/mac-3.99-u4-b5/src/Shared/StdLibFileIO.cpp	2006-08-06 18:34:19.993750000 -0500
@@ -8,7 +8,7 @@
 ///////////////////////////////////////////////////////
 
 // low level I/O, where are prototypes and constants?
-#if   defined _WIN32  ||  defined __TURBOC__  ||  defined __ZTC__  ||  defined _MSC_VER
+#if   defined _WIN32  ||  defined __TURBOC__  ||  defined __ZTC__  ||  defined _MSC_VER || defined __CYGWIN__
 # include <io.h>
 # include <fcntl.h>
 # include <time.h>
